### PR TITLE
Add internal solution for Distributing Apples (CSES-1716)

### PR DESCRIPTION
**Pull request by ChuangSheep**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making the changes.

### CODE REVIEW

File: solutions/gold/cses-1716.mdx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+source: CSES
+title: Distributing Apples
+author: Chuyang Wang
+---
+
+## Explanation
+
+To get a more illustrative idea of the given problem, consider $n + m - 1$ white
+balls in a row. Of those $n + m - 1$ white balls, $m - 1$ are chosen to be
+colored black as separators, which gives us exactly $m$ segments of (possibly
```
Comment (SansPapyrus683) [inline solutions/gold/cses-1716.mdx:12]: maybe link to a resource abt stars & bars here

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/gold/cses-1716.mdx [hunk 2]
```diff
@@ -70,8 +71,14 @@ ll qexp(ll base, ll exp) {
...
 
-void precalc() {
-	fac[0] = fac[1] = 1;
+ll binom(int n, int k) {
+	return fac[n] * inv[k] % MOD * inv[n - k] % MOD;
+}
+
+int main() {
+	fac.assign(MAXN, 1);
+	inv.assign(MAXN, 1);
```
Comment (SansPapyrus683) [inline solutions/gold/cses-1716.mdx:80]: you can just initialize this globally i think
```cpp
vector<ll> fac(MAXN, 1);
```

File: solutions/gold/cses-1716.mdx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+using ll = long long;
+
+const ll MOD = 1e9 + 7;
+const int MAXN = 2e6;
+
+vector<ll> fac(MAXN, 1);
+vector<ll> inv(MAXN, 1);
+
+ll qexp(ll base, ll exp) {
```
Comment (SansPapyrus683) [inline solutions/gold/cses-1716.mdx:59]: can you use the guide's bin exp and send a link
consistency is always good

Comment (ChuangSheep) [reply solutions/gold/cses-1716.mdx:59]: which one do you prefer? cause there are multiple different versions here https://usaco.guide/gold/combo?lang=cpp

Comment (SansPapyrus683) [reply solutions/gold/cses-1716.mdx:59]: i was thinking about the one [here](https://usaco.guide/gold/modular?lang=cpp)...

**Review by SansPapyrus683 — APPROVED** [review]
@ryanchou-dev

File: solutions/gold/cses-1716.mdx [hunk 4]
```diff
@@ -68,9 +68,7 @@ ll binpow(ll x, ll n, ll m) {
 	return res;
 }
 
-ll binom(int n, int k) {
-	return fac[n] * inv[k] % MOD * inv[n - k] % MOD;
-}
+ll binom(int n, int k) { return fac[n] * inv[k] % MOD * inv[n - k] % MOD; }
```
Comment (SansPapyrus683) [inline solutions/gold/cses-1716.mdx:71]: bruh.

Comment (ChuangSheep) [reply solutions/gold/cses-1716.mdx:69]: lol guess we're having a code convention that's being enforced automatically

**Review by ryanchou-dev — APPROVED** [review]
ok slay

**Pull request merged**